app/services/oneoffs/npq/set_special_course_for_npq_contracts.rb

```diff
@@ -10,28 +10,35 @@ def initialize(cohort_year:, payment_date_range:, course_identifier:)
 
     def call
       ActiveRecord::Base.transaction do
-        statements.each do |statement|
-          npq_lead_provider = statement.npq_lead_provider
-          version = statement.contract_version
+        old_contracts = []
```

Comment:
thanks for fixing this, any chance we can update the specs for this service to ensure we don't miss anything? Also there are failures in the specs from I think the versioning in factories, happy to have a look together/revert if its too gnarly :)